### ISSUE TITLE
CustomImportOrder: should check empty line separator between import statements but not the ones before import statement

### ISSUE THREAD

**Issue by gmshake** [checkstyle/checkstyle#7314]
https://checkstyle.org/config_imports.html#CustomImportOrder

```
/var/tmp $ javac CustomImportOrder.java
```

```
/var/tmp $ cat config.xml
```
```xml
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
 "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">

    <module name="CustomImportOrder">
      <property name="customImportOrderRules"
               value="STATIC###STANDARD_JAVA_PACKAGE"/>
    </module>

  </module>
</module>
```
```
/var/tmp $ cat CustomImportOrder.java
```
```java
package com.example;

import static java.awt.Button.ABORT;
import static java.io.File.createTempFile;
import static javax.swing.WindowConstants.*;

// comment between import groups

import java.io.File; // should warn
import java.io.IOException;


// comment within import group
import java.io.InputStream; // should warn
import java.io.Reader;

// comment within import group

import javax.swing.JTable; // should warn

class CustomImportOrder {
}
```
```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.27-all.jar -c config.xml CustomImportOrder.java
Starting audit...
Audit done.
```

---------------

The lines in `CustomImportOrder.java` commented with `should warn` should trigger violations during checkstyle check.
The comments between import statements should take account into since we're checking empty lines between import statements but not the ones before import statement.

--------------

**Comment by romani** [comment]
While CustomImportOrder in documentation does not mention anything about empty lines inside group , so we can treat it as gray area.
This check was created specifically for Google style.
https://checkstyle.org/google_style.html#a3.3.3
now Google style clearly mention that `If there are both static and non-static imports, a single blank line separates the two blocks. There are no other blank lines between import statements.`
https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s3.3.3-import-ordering-and-spacing .

So we need to update Check forbid extra empty lines inside import groups.

NOTE: it might in future somebody request to make it controlled by property, so we will do this separately, and default behavior will be no empty lines inside group.

**Comment by romani** [comment]
I am about to approve this issue.
@rnveach  and @pbludov , please review and share your thoughts on approval.

**Comment by pbludov** [comment]
The check should be in sync with the Google style.

I'm not sure about property, because nobody writes imports manually. Today they are managed by the IDE. When the IDE changes an import, such comments will at best be moved to a random location, and at worst deleted altogether.

**Comment by romani** [comment]
I presume @pbludov is ok to approve issue.

**Comment by pbludov** [comment]
> I presume @pbludov is ok to approve issue.

Yes I'm ok.

**Comment by romani** [comment]
@gmshake , How is this issue different from #7278 ?

**Comment by gmshake** [comment]
@romani  I might mislead you as the `CustomImportOrder.java` is much like the one in https://github.com/checkstyle/checkstyle/issues/7278 .

To be clear, this issue is about comments between import statements only. For https://github.com/checkstyle/checkstyle/issues/7278, it is about misleading violation message when checking empty line separator within import group.

**Comment by gmshake** [comment]
Once #6680 was fixed, I will file PRs and show more accurately the difference.

**Comment by romani** [comment]
behavior is the same(no violations at all) on today master build.

**Comment by rnveach** [comment]
https://gmshake.github.io/checkstyle_contribution/checkstyle-tester/diff_pr_7365_4/checkstyle/index.html#A1
https://gmshake.github.io/checkstyle_contribution/checkstyle-tester/diff_pr_7365_4/checkstyle/xref/Users/zlei/Downloads/checkstyle_contribution/checkstyle-tester/repositories/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/customimportorder/InputCustomImportOrder_OverlappingPatterns.java.html#L13
How come there is a violation on line 13 but not line 9?

**Comment by rnveach** [comment]
Fix was merged
